src/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanner.java

```diff
@@ -60,14 +61,24 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 				for (Element sourceElement : sourceElements) {
 					String src = sourceElement.getAttributeValue("src");
 						if (src != null && isScriptFromOtherDomain(msg.getRequestHeader().getHostName(), src, msg)) {
-							this.raiseAlert(msg, id, src);
+							String integrity = sourceElement.getAttributeValue("integrity");
+							if (integrity != null && integrity.trim().length() > 0) {
+								/*
+								 * If it has an integrity value assume its fine
+								 * We dont check the integrity value is valid because
+								 * 1. pscan rules cant make new requests and
+								 * 2. the browser will check it anyway
+								 */
+								continue;

```

Comment:
Might be more clearn if the logic is reversed here and the alert raising moved in. Or if you really like this the comment should make it clear that the continue moves to the next sourceElement because integrity was found on the current one.